test/test_mixed_null_assertions.cpp

```diff
@@ -78,3 +78,253 @@ TEST(List_Mixed_do_insert)
     list.insert_null(0);
     list.insert(0, Mixed("hello world"));
 }
+
+TEST(Mixed_List_unresolved_as_null)
+{
+    Group g;
+    auto t = g.add_table("foo");
+    t->add_column_list(type_Mixed, "mixeds");
+    auto obj = t->create_object();
+    auto obj1 = t->create_object();
+
+    auto list = obj.get_list<Mixed>("mixeds");
+
+    list.insert_null(0);
+    list.insert(1, Mixed{"test"});
+    list.insert(2, obj1);
+    obj1.invalidate();
+
+    CHECK_EQUAL(list.size(), 3);
+
+    {
+        // find all mixed nulls or unresolved link should work the same way
+        std::vector<size_t> found;
+        auto check_results = [&](std::vector<size_t> expected) -> bool {
+            if (found.size() != expected.size())
+                return false;
+
+            std::sort(found.begin(), found.end());
+            std::sort(expected.begin(), expected.end());
+            for (size_t i = 0; i < found.size(); ++i) {
+                if (found[i] != expected[i])
+                    return false;
+            }
+            return true;
+        };
+        list.find_all(realm::null(), [&](size_t pos) {
+            found.push_back(pos);
+        });
+        CHECK_EQUAL(check_results({0, 2}), true);
+    }
+
+    {
+        // find null or find unresolved link diverge, different objects should be returned
+        auto index = list.find_any(realm::null());
+        CHECK_EQUAL(index, 0);
+        index = list.find_first(obj1);
+        CHECK_EQUAL(index, 2);
+        // but both should look like nulls
+        CHECK_EQUAL(list.is_null(0), true);
```

Comment:
Ok, I did not think you would change the cases where we check a boolean. Here we just use CHECK. You can decide for yourself if you want to change back.